### PR TITLE
feat: Add pagination to footer in 'full-page' variants for table and cards

### PR DESCRIPTION
**Pull request by rubencarvalho**
### Description

On small viewports, when pagination is in the header of the page (due to the sticky header being deactivate to conserve space), to access the pagination users have to scroll to the top of the page.
This PR addresses this by implementing a more discoverable pagination in the footer of `full-page` variants of the table and cards. We keep this feature for both desktop and mobile viewports for consistency.

Designs doc: DOX3AXbqOZ6i

### How has this been tested?

Added test utils and respective units tests.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test-utils/dom/cards/index.ts [hunk 1]
```diff
@@ -86,6 +86,10 @@ export default class CardsWrapper extends ComponentWrapper {
     return this.findComponent(`.${tableStyles['tools-pagination']}`, PaginationWrapper);
   }
 
+  findFooterPagination(): PaginationWrapper | null {
+    return this.findComponent(`.${styles['footer-pagination']}`, PaginationWrapper);
+  }
```
Comment (just-boris) [inline src/test-utils/dom/cards/index.ts:91]: What is the point of making this util public?

**Review by just-boris — CHANGES_REQUESTED** [review]
Needs a visual regression test for pagination + footer case. 

This page is a good fit for it: https://github.com/cloudscape-design/components/blob/main/pages/table/permutations.page.tsx

File: src/table/internal.tsx [hunk 2]
```diff
@@ -271,11 +274,14 @@ const InternalTable = React.forwardRef(
           __disableFooterDivider={true}
           __disableStickyMobile={false}
           footer={
-            footer && (
+            hasFooter ? (
               <div className={clsx(styles['footer-wrapper'], styles[`variant-${computedVariant}`])}>
-                <div className={styles.footer}>{footer}</div>
+                <div className={clsx(styles.footer, hasMobilePagination && styles['footer-with-pagination'])}>
+                  {footer && <span>{footer}</span>}
+                  {hasMobilePagination && <div className={styles['footer-pagination']}>{pagination}</div>}
```
Comment (just-boris) [inline src/table/internal.tsx:281]: Typically, `footer` slot contains center-aligned text. How does this line up when pagination adds? Could you share the design specs in DM?

File: src/test-utils/dom/cards/index.ts (hunk 1 above)
Comment (rubencarvalho) [reply src/test-utils/dom/cards/index.ts]: I was also not sure about its utility, but your comment is just another datapoint to confirm it. I removed it.

File: src/test-utils/dom/table/index.ts [hunk 3]
```diff
@@ -141,6 +141,10 @@ export default class TableWrapper extends ComponentWrapper {
     return this.findComponent(`.${styles['tools-pagination']}`, PaginationWrapper);
   }
 
+  findFooterPagination(): PaginationWrapper | null {
```
Comment (just-boris) [inline src/test-utils/dom/table/index.ts:144]: Removed for cards already, but should be removed for the table too

**Review by just-boris — APPROVED** [review]

**Pull request merged**